nncf/torch/extensions/src/quantization/cuda/functions_cuda_impl.cu

```diff
@@ -452,14 +494,25 @@ std::vector<at::Tensor> q_scale_per_activation_channel_cuda_backward(at::Tensor
     auto dev_last_block_counter_range = at::zeros({grid_size.x, 1},  at::device(grad_output.options().device()).dtype(at::kInt));
     auto dev_last_block_counter_low = at::zeros({grid_size.x, 1},  at::device(grad_output.options().device()).dtype(at::kInt));
 
+    const auto grad_input_dtype = grad_output.scalar_type();
+    if (grad_input_dtype == c10::ScalarType::Half)
+    {
+        grad_input_range = grad_input_range.toType(c10::ScalarType::Float);
+        grad_input_low = grad_input_low.toType(c10::ScalarType::Float);
+        dev_tmp_range = dev_tmp_range.toType(c10::ScalarType::Float);
+        dev_tmp_low = dev_tmp_low.toType(c10::ScalarType::Float);
```

Comment:
Looks like casting is redundant here. I think we can create dev_tmp_range and dev_tmp_low tensors using accumulator type, but grad_input_low and grad_input_range tensors using output/input type.

Reply:
Agree with you, thank you for the suggestion! I implemented approach you described, please check